views/campgrounds/index.ejs

```diff
@@ -1,159 +1,166 @@
+                <input class="campground-search-input" type="search" placeholder="Search for campsites..."
+                    aria-label="Search" name="q" id="search-input">
+                <button class="campground-search-button" type="submit" id="search-button">Search</button>
             </div>
+        </form>
+
+        <!-- Sort Controls -->
+        <div class="campground-sort-controls">
+            <button id="show-nearest-btn" class="campground-nearest-btn">Show Nearest to Me</button>
+            <form action="/campgrounds" method="GET" class="d-inline">
+                <select name="sort" id="sort" class="campground-sort-select">
+                    <option value="">--Select--</option>
+                    <option value="priceAsc" <%=(locals.query && query.sort==='priceAsc' ) ? 'selected' : '' %>>Price:
+                        Low
+                        to High</option>
+                    <option value="priceDesc" <%=(locals.query && query.sort==='priceDesc' ) ? 'selected' : '' %>>Price:
+                        High to Low</option>
+                    <option value="reviewsHighest" <%=(locals.query && query.sort==='reviewsHighest' ) ? 'selected' : ''
+                        %>
+                        >Reviews: Highest Rated</option>
+                    <option value="reviewsLowest" <%=(locals.query && query.sort==='reviewsLowest' ) ? 'selected' : ''
+                        %>
+                        >Reviews: Lowest Rated</option>
+                </select>
+                <button type="submit" class="campground-sort-button">Sort</button>
+            </form>
         </div>
-    </form>
-</div>
-
-<!-- Campgrounds List -->
-<div id="campground-container">
-    <% for(let campground of campgrounds){ %>
-        <div class="card text-white bg-dark mb-3" id="campinfo">
-            <div class="row">
-                <div class="col-md-4" id="campgroundhomeimage">
+    </div>
+
+    <div class="campground-listing-grid">
+        <% for(let campground of campgrounds){ %>
+            <div class="campground-card">
+                <div class="campground-image-container">
                     <% if(campground.images.length) { %>
-                        <img crossorigin="anonymous" id="indeximg" alt="" src="<%= campground.images[0].url %>">
-                    <% } else { %>
-                        <img crossorigin="anonymous" class="img-fluid" src="https://res.cloudinary.com/dgr21eiov/image/upload/v1723129452/CampX/eajljfmhshlozrm5yams.jpg" alt="">
-                    <% } %>
+                        <img class="campground-image" alt="" src="<%= campground.images[0].url %>"
+                            crossorigin="anonymous">
+                        <% } else { %>
+                            <img class="campground-image"
+                                src="https://res-console.cloudinary.com/dgr21eiov/media_explorer_thumbnails/149d48129829d71b6f4ffd8347709f20/detailed"
```

Comment:
replace the src with this url ``https://res.cloudinary.com/dgr21eiov/image/upload/v1723129452/CampX/eajljfmhshlozrm5yams.jpg``